src/client.ts

```diff
@@ -993,17 +994,14 @@ export class Client {
           parsed.responseCode === responseCodes.RESP_ERR_WRONG_WALLET;
         const invalidEphemId =
           parsed.responseCode === responseCodes.RESP_ERR_INVALID_EPHEM_ID;
-        const success = 
-          parsed.responseCode === responseCodes.RESP_SUCCESS && 
-          !parsed.err;
         const canRetry = retryCount > 0;
         // Re-connect and/or retry request if needed
         if (canRetry && deviceBusy) {
           // Wait a few seconds and retry
           setTimeout(() => {
             this._request(payload, encReqCode, cb, retryCount - 1);
           }, 3000);
-        } else if (canRetry && (wrongWallet || invalidEphemId)) {
+        } else if (canRetry && ((wrongWallet && this._canReplaceWalletUID()) || invalidEphemId)) {
```

Comment:
Should `_canReplaceWalletUID()` have an argument?

Reply:
fixed in force push